### ISSUE TITLE
unable to mount shared directories

### ISSUE THREAD

**Issue by mardif** [dotless-de/vagrant-vbguest#326]
windows 10
vagrant 2.2.3
virtualbox 6.0.2
vbguest 6.0.2

D:\projects\vagrant>vagrant reload
==> default: Attempting graceful shutdown of VM...
==> default: Checking if box 'centos/7' version '1812.01' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: You are trying to forward to privileged ports (ports <= 1024). Most
==> default: operating systems restrict this to only privileged process (typically
==> default: processes running as an administrative user). This is a warning in case
==> default: the port forwarding doesn't work. If any problems occur, please try a
==> default: port higher than 1024.
==> default: Forwarding ports...
    default: 80 (guest) => 80 (host) (adapter 1)
    default: 443 (guest) => 443 (host) (adapter 1)
    default: 3306 (guest) => 3306 (host) (adapter 1)
    default: 5432 (guest) => 5432 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
    default: Warning: Connection aborted. Retrying...
==> default: Machine booted and ready!
[default] GuestAdditions 6.0.2 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Rsyncing folder: /cygdrive/d/projects/vagrant/ => /vagrant
==> default: Mounting shared folders...
    default: /var/www/html/project=> D:/projects/project
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o dmode=775,fmode=664,uid=1000,gid=48 project /var/www/html/project

The error output from the command was:

: Protocol error

**Comment by mardif** [comment]
I tried to install previous version of guestaddition (6.0.0), but nothing changes.
So, I tried to re-install vbox guestadditions, I goy this log:

D:\projects\vagrant>vagrant reload
==> default: Attempting graceful shutdown of VM...
==> default: Checking if box 'centos/7' version '1812.01' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: You are trying to forward to privileged ports (ports <= 1024). Most
==> default: operating systems restrict this to only privileged process (typically
==> default: processes running as an administrative user). This is a warning in case
==> default: the port forwarding doesn't work. If any problems occur, please try a
==> default: port higher than 1024.
==> default: Forwarding ports...
    default: 80 (guest) => 80 (host) (adapter 1)
    default: 443 (guest) => 443 (host) (adapter 1)
    default: 3306 (guest) => 3306 (host) (adapter 1)
    default: 5432 (guest) => 5432 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
    default: Warning: Connection reset. Retrying...
==> default: Machine booted and ready!
[default] GuestAdditions versions on your host (6.0.2) and guest (6.0.0) do not match.
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: it.centos.contactlab.it
 * extras: it.centos.contactlab.it
 * remi-safe: mirror.reconn.ru
 * updates: it.centos.contactlab.it
Package kernel-devel-3.10.0-957.1.3.el7.x86_64 already installed and latest version
Package kernel-devel-3.10.0-957.1.3.el7.x86_64 already installed and latest version
Package gcc-4.8.5-36.el7.x86_64 already installed and latest version
Package binutils-2.27-34.base.el7.x86_64 already installed and latest version
Package 1:make-3.82-23.el7.x86_64 already installed and latest version
Package 4:perl-5.16.3-293.el7.x86_64 already installed and latest version
Package bzip2-1.0.6-13.el7.x86_64 already installed and latest version
Nothing to do
Copy iso file C:\Program Files\Oracle\VirtualBox\VBoxGuestAdditions.iso into the box /tmp/VBoxGuestAdditions.iso
Mounting Virtualbox Guest Additions ISO to: /mnt
mount: /dev/loop0 is write-protected, mounting read-only
Installing Virtualbox Guest Additions 6.0.2 - guest version is 6.0.0
Verifying archive integrity... All good.
Uncompressing VirtualBox 6.0.2 Guest Additions for Linux........
VirtualBox Guest Additions installer
Removing installed version 6.0.0 of VirtualBox Guest Additions...
Copying additional installer modules ...
Installing additional modules ...
VirtualBox Guest Additions: Building the VirtualBox Guest Additions kernel modules.  This may take a while.
VirtualBox Guest Additions: To build modules for other installed kernels, run
VirtualBox Guest Additions:   /sbin/rcvboxadd quicksetup <version>
VirtualBox Guest Additions: Building the modules for kernel 3.10.0-957.1.3.el7.x86_64.
VirtualBox Guest Additions: Running kernel modules will not be replaced until the system is restarted
VirtualBox Guest Additions: Starting.
An error occurred during installation of VirtualBox Guest Additions 6.0.2. Some functionality may not work as intended.
In most cases it is OK that the "Window System drivers" installation failed.
Redirecting to /bin/systemctl start vboxadd.service
Redirecting to /bin/systemctl start vboxadd-service.service
Unmounting Virtualbox Guest Additions ISO from: /mnt
Got different reports about installed GuestAdditions version:
Virtualbox on your host claims:   6.0.0
VBoxService inside the vm claims: 6.0.2
Going on, assuming VBoxService is correct...
Got different reports about installed GuestAdditions version:
Virtualbox on your host claims:   6.0.0
VBoxService inside the vm claims: 6.0.2
Going on, assuming VBoxService is correct...
==> default: Checking for guest additions in VM...
==> default: Rsyncing folder: /cygdrive/d/projects/vagrant/ => /vagrant
==> default: Mounting shared folders...
    default: /var/www/html/project=> D:/projects/project
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o dmode=775,fmode=664,uid=1000,gid=48 project /var/www/html/project

The error output from the command was:

/sbin/mount.vboxsf: mounting failed with the error: Protocol error

**Comment by g24swint** [comment]
I am seeing this same error as well, and I would further note that the message actually says the Additions are running "OK".

Host: Windows 10
Guest: ubuntu/bionic64
Vbox: 6.0.4
VBoxGuestAdditions were originally a 5.x version (can't remember which) and the plug updated the additions to be 6.0.4.

The first time running, when the plugin initially kicked in to install the new VBoxGuestAdditions, the shared folder mappings all actually worked.

Later, after restarting the VirtualBox and Vagrant Image both (sorry, not sure if it's dependent on one or the other), the vagrant image started issuing the "Protocol error" as given above.

I tried a few combinations of `sudo  apt-get installs` of VBox packages to see if I could force the requisite settings into place, but no luck.

Startup output:

```
F:\vagrant\ubuntu1804>vagrant up
Bringing machine 'default' up with 'virtualbox' provider...
==> default: Checking if box 'ubuntu/bionic64' version '20190212.1.0' is up to date...
==> default: Clearing any previously set forwarded ports...
==> default: Clearing any previously set network interfaces...
==> default: Preparing network interfaces based on configuration...
    default: Adapter 1: nat
==> default: Forwarding ports...
    default: 5432 (guest) => 5432 (host) (adapter 1)
    default: 22 (guest) => 2222 (host) (adapter 1)
==> default: Running 'pre-boot' VM customizations...
==> default: Booting VM...
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
==> default: Machine booted and ready!
[default] GuestAdditions 6.0.4 running --- OK.
==> default: Checking for guest additions in VM...
==> default: Mounting shared folders...
    default: /vagrant => F:/vagrant/ubuntu1804
Vagrant was unable to mount VirtualBox shared folders. This is usually
because the filesystem "vboxsf" is not available. This filesystem is
made available via the VirtualBox Guest Additions and kernel module.
Please verify that these guest additions are properly installed in the
guest. This is not a bug in Vagrant and is usually caused by a faulty
Vagrant box. For context, the command attempted was:

mount -t vboxsf -o uid=1000,gid=1000 vagrant /vagrant

The error output from the command was:

: Protocol error

F:\vagrant\ubuntu1804>
```

Similarly, I get an error when attempting to run the mount command via `sudo`

```bash
F:\vagrant\ubuntu1804>vagrant ssh
Welcome to Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-45-generic x86_64)Welcome to Ubuntu 18.04.2 LTS (GNU/Linux 4.15.0-45-generic x86_64)

 * Documentation:  https://help.ubuntu.com
 * Management:     https://landscape.canonical.com
 * Support:        https://ubuntu.com/advantage

  System information as of Sun Feb 17 23:49:16 UTC 2019

  System load:  0.0                Processes:             103
  Usage of /:   12.4% of 96.88GB   Users logged in:       0
  Memory usage: 13%                IP address for enp0s3: 10.0.2.15
  Swap usage:   0%

[--- CUT UBUNTU MESSAGES ---]

0 packages can be updated.
0 updates are security updates.


Last login: Sun Feb 17 22:27:14 2019 from 10.0.2.2
vagrant@ubuntu-bionic:~$ sudo mount -t vboxsf -o uid=1000,gid=1000 vagrant /vagrant
/sbin/mount.vboxsf: mounting failed with the error: Protocol error
```

**Comment by g24swint** [comment]
Too add to the frustration, I attempted to reproduce with a `vagrant destroy` / `vagrant up` process to start anew. I now have brought up the VM several times, even with a full host restart, and can't reproduce the error.

I did clearly see the vbguest plugin fire-off and update the VBoxGuestAdditions.

Potentially something is getting corrupted in the course of the machine running that keeps it from re-mounting the shared folders.
